controllers/vrg_volrep.go

```diff
@@ -1466,6 +1466,18 @@ func setPVCClusterDataProtectedCondition(protectedPVC *ramendrv1alpha1.Protected
 	}
 }
 
+// ensureVRDeletedFromAPIServer adds an additional step to ensure that we wait for volumereplication deletion
+// from API server before moving ahead vrg finalizer removal.
+func (v *VRGInstance) ensureVRDeletedFromAPIServer(vrNamespacedName types.NamespacedName) error {
+	if err := v.reconciler.APIReader.Get(v.ctx, vrNamespacedName, v.instance); err != nil {
```

Comment:
v.instance is of type VolumeReplicationGroup, we are getting VolumeReplication resource here, not the group, and this would overwrite the instance based on the new get as well.